package.json

```
@@ -26,27 +28,42 @@
     "usehooks-ts": "^3.1.0"
   },
   "devDependencies": {
-    "@next/eslint-plugin-next": "^14.2.11",
+    "@eslint/compat": "^1.2.7",
+    "@eslint/js": "^9.23.0",
+    "@next/eslint-plugin-next": "^15.2.4",
     "@svgr/webpack": "8.1.0",
     "@types/node": "^20.12.2",
     "@types/react": "^18.3.4",
     "@types/react-dom": "^18.3.0",
     "@typescript-eslint/eslint-plugin": "^8.2.0",
     "@typescript-eslint/parser": "^8.2.0",
     "dotenv": "^10.0.0",
-    "eslint": "^8.39.0",
+    "eslint": "^9.23.0",
     "eslint-config-prettier": "^9.1.0",
     "eslint-plugin-prettier": "^5.2.1",
-    "eslint-plugin-react": "^7.32.2",
-    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-react": "^7.37.4",
+    "eslint-plugin-react-hooks": "^5.2.0",
     "eslint-plugin-simple-import-sort": "^12.1.1",
     "eslint-plugin-unused-imports": "^4.1.3",
+    "globals": "^16.0.0",
     "google-auth-library": "^9.14.0",
     "google-spreadsheet": "^4.1.2",
+    "husky": "^9.1.7",
+    "lint-staged": "^15.5.0",
     "prettier": "^3.3.3",
     "sass": "1.77.8",
     "typescript": "^5.5.4",
+    "typescript-eslint": "^8.28.0",
     "webpack": "5.93.0"
   },
+  "lint-staged": {
+    "*.{ts,tsx,js,jsx}": [
```

```suggestion
    "*.{ts,tsx,js,cjs,mjs,jsx}": [
```

확장하여 `cjs`, `mjs`도 추가하는 것은 어떨까요?